tests/operators/big_int.cpp

```diff
@@ -0,0 +1,519 @@
+             result.insert(0,1,((mul%10)+'0'));
+              mul=mul/10;
+               //cout<<s<<endl;
+          }
+          if(mul)
+            result.insert(0,1,(mul+'0'));
+          s[i]=result;
+          //cout<<s[i]<<endl;
+}
+int j=0;
+
+for(int i=l2-1;i>=0;i--)
+{
+  string temp=s[str2[i]-'0'];
+  //cout<<temp<<" ";
+  temp.insert(temp.end(),l2-1-i,'0');
+  //cout<<res<<" ";
+  //BigInteger temp1(temp);
+  res=res+temp;
+  //cout<<temp1<<" "<<res<<endl;
+j++;
+}
+return res;}
+
+            
+              }*/
+
+
+/*BigInteger operator* (BigInteger  a, BigInteger  b) {
+
+BigInteger res("0");
+string str1=a.str;
+string str2=b.str;
+string s[10];
+s[0]="0";
+s[1]=str1;
+
+int l=str1.size()-1;
+for(int i=2;i<=9;i++)
+{ string result;
+  int mul=0;
+for (int j=l;j>=0;j--) {
+            mul += i*(str1[j]-'0');
+            //cout<<s<<endl;
+             result.insert(0,1,((mul%10)+'0'));
+              mul=mul/10;
+               //cout<<s<<endl;
+          }
+          if(mul)
+            result.insert(0,1,(mul+'0'));
+          s[i]=result;
+          //cout<<s[i]<<endl;
+}
+int j=0;
+for(int i=str2.size()-1;i>=0;i--)
+{
+  string temp=s[str2[i]-'0'];
+  //cout<<temp<<" ";
+  temp=temp.append(j,'0');
+  //cout<<res<<" ";
+  //BigInteger temp1(temp);
+  res=res+temp;
+  //cout<<temp1<<" "<<res<<endl;
+j++;
+}
+return res;
+}*/
+
+/*
+BigInteger operator/ (BigInteger  a, BigInteger  b) {
+        //string s="";
+        BigInteger div("0");
+        if(a.sign==1 && b.sign==1){ }
+       else if(a.sign ==1){ div.sign=1;}
+      else if(b.sign ==1){ div.sign=1;}
+     //if(b>a){return b/a;} 
+       
+       //BigInteger c=a;
+       // cout<<c<<" "<<b<<" "<<endl;
+       while( a >= b )
+       {
+        //bool bl = c>=b ;
+          a=a-b;
+          div=div+1;
+          cout<<a<<" "<<b<<endl;
+        }
+
+           return div;
+            
+              }
+*/
+
+BigInteger operator% (BigInteger  a, BigInteger  b) {
+        //string s="";
+
+        BigInteger mod("0");
+      string str1=b.str;
+      int si=str1.size();
+      BigInteger temp("0");
+      int diff= a.str.size()-si;
+      while(diff>2)
+      { 
+        string str (diff,'1');
+      BigInteger b1(str);
+      temp=b*b1;
+        a=a-temp;
+       diff= a.str.size()-si;
+       }
+   while(a<0)
+        {
+          a=a+b;
+        }
+while( a >= b )
+       {
+        //bool bl = c>=b ;
+          a=a-b;
+         //cout<<a<<" "<<b<<endl;
+        }
+
+           return a;
+              }
+
+ostream & operator<< (ostream &out, const BigInteger &a) {
+      string s=a.str;
+      
+      if(a.sign==1)
+        {
+         
+          s='-'+s;
+        }out<<s;
+    return out;
+}
+BigInteger  factorial(int a)
+   { BigInteger i("2");
+   BigInteger ans("1");
+  // cout<<a;
+   for(;i<=a;i=i+1)
+   {
+    
+      ans=ans*i;
+      //cout<<ans<<endl; 
+   }
+    
+return ans;
+
+   }
+BigInteger expo(BigInteger a, ll b){
+  BigInteger e("1");
```

Comment:
Codacy found an issue: [Variable 'e' is assigned a value that is never used.](https://app.codacy.com/gh/ayaankhan98/bigINT/pullRequest?prid=6290647)